# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [2]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [10]:
Series1 = pd.Series(np.random.randint(1 , 5 , 100 , dtype='l'))
Series2 = pd.Series(np.random.randint(1 , 4 , 100 , dtype='l'))
Series3 = pd.Series(np.random.randint(10000 , 30001 , 100 , dtype='l'))

### Step 3. Let's create a DataFrame by joinning the Series by column

In [11]:
house = pd.concat([Series1,Series2 ,Series3] , axis=1)
house.head()

0  1      2
0  1  1  17315
1  4  3  27185
2  3  2  11305
3  4  1  26572
4  1  1  22681

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [13]:
house.rename(columns={0 : 'bedrs' , 1 : 'barths' , 2 : 'price_sqr_meter'} , inplace=True)
house.head()

bedrs  barths  price_sqr_meter
0      1       1            17315
1      4       3            27185
2      3       2            11305
3      4       1            26572
4      1       1            22681

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [18]:
bigcolumn =pd.concat([Series1 , Series2 , Series3],  axis=0)
bigcolumn = bigcolumn.to_frame()
bigcolumn.info()

<class 'pandas.core.frame.DataFrame'>
Index: 300 entries, 0 to 99
Data columns (total 1 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   0       300 non-null    int32
dtypes: int32(1)
memory usage: 3.5 KB


In [21]:
bigcolumn

0
0        1
1        4
2        3
3        4
4        1
..     ...
295  11878
296  11835
297  25141
298  22156
299  15227

[300 rows x 1 columns]

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [19]:
len(bigcolumn)

300

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [20]:
bigcolumn.reset_index(drop=True , inplace=True)
bigcolumn

0
0        1
1        4
2        3
3        4
4        1
..     ...
295  11878
296  11835
297  25141
298  22156
299  15227

[300 rows x 1 columns]